cardano-api/src/Cardano/Api/Typed.hs

```diff
@@ -1606,78 +1609,70 @@ instance FromJSON MultiSigScript where
 parseMultiSigScript :: HMS.HashMap Text Value -> MultiSigScript
 parseMultiSigScript hms = case  all' hms <|> any' hms <|> atLeast hms of
                             Just mss -> mss
-                            Nothing -> error "No multisig script found"
+                            Nothing -> error "No multi-signature script found"
  where
   convertValueToMultiSigScript :: Value -> Maybe MultiSigScript
-  convertValueToMultiSigScript (String pkH) = Just . RequireSignature $ convertToHash pkH
   convertValueToMultiSigScript (Object valueHm) =
-    any' valueHm  <|> all' valueHm <|> atLeast valueHm
+     payKeyHash' valueHm <|> any' valueHm  <|> all' valueHm <|> atLeast valueHm
   convertValueToMultiSigScript _ = Nothing
 
+  payKeyHash' :: HMS.HashMap Text Value -> Maybe MultiSigScript
+  payKeyHash' hm = case HMS.lookup "type" hm of
+                    Just (String "requireSignature") ->
+                      case HMS.lookup "paymentKeyHash" hm of
+                        Just (String pkh) -> Just . RequireSignature $ convertToHash pkh
+                        Just val -> error $ "\"paymentKeyHash\" value is not a payment key hash. Value: " <> show val
+                        Nothing -> error "\"paymentKeyHash\" field is empty in your multi-signature script."
```

Comment:
I initially put things in the `Maybe` monad because `Parser`'s `Alternative` instance was not behaving as desired. The `Parser` would succeed only on whatever was first i.e:
` payKeyHash' valueHm <|> any' valueHm  <|> all' valueHm <|> atLeast valueHm` would succeed in parsing `any` scripts only
` payKeyHash' valueHm <|> all' valueHm <|> any' valueHm  <|>  atLeast valueHm`
would succeed in parsing `all` scripts only
etc

I've found a way around this however.